src/ophyd_async/epics/areadetector/aravis.py

```diff
@@ -0,0 +1,70 @@
+from typing import get_args
+
+from bluesky.protocols import HasHints, Hints
+
+from ophyd_async.core import DirectoryProvider, StandardDetector, TriggerInfo
+from ophyd_async.epics.areadetector.controllers.aravis_controller import (
+    AravisController,
+)
+from ophyd_async.epics.areadetector.drivers import ADBaseShapeProvider
+from ophyd_async.epics.areadetector.drivers.aravis_driver import AravisDriver
+from ophyd_async.epics.areadetector.writers import HDFWriter, NDFileHDF
+
+
+class AravisDetector(StandardDetector, HasHints):
+    """
+    Ophyd-async implementation of an ADAravis Detector.
+    The detector may be configured for an external trigger on a GPIO port,
+    which must be done prior to preparing the detector
+    """
+
+    _controller: AravisController
+    _writer: HDFWriter
+
+    def __init__(
+        self,
+        prefix: str,
+        name: str,
+        directory_provider: DirectoryProvider,
+        driver: AravisDriver,
+        hdf: NDFileHDF,
+        gpio_number: AravisController.GPIO_NUMBER = 1,
+        **scalar_sigs: str,
+    ):
+        # Must be child of Detector to pick up connect()
+        self.drv = driver
+        self.hdf = hdf
+
+        super().__init__(
+            AravisController(self.drv, gpio_number=gpio_number),
+            HDFWriter(
+                self.hdf,
+                directory_provider,
+                lambda: self.name,
+                ADBaseShapeProvider(self.drv),
+                **scalar_sigs,
+            ),
+            config_sigs=(self.drv.acquire_time, self.drv.acquire),
+            name=name,
+        )
+
+    async def _prepare(self, value: TriggerInfo) -> None:
+        await self.drv._fetch_deadtime()
```

Comment:
Can we do this without private access? To be honest could just make `_fetch_deadtime()` a public method

Reply:
Or make a property